pkg/arvo/app/invite-store.hoon

```diff
@@ -0,0 +1,186 @@
+/+  *invite-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff diff]
+      [%quit ~]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  =invites
+  ==
+::
++$  diff
+  $%  [%invite-initial invites]
+      [%invite-update invite-update]
+  ==
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    [~ this]
+  [~ this(+<+ u.old)]
+::
+++  peek-x-all
+  |=  pax=path
+  ^-  (unit (unit [%noun (map path invitatory)]))
+  [~ ~ %noun invites]
+::
+++  peek-x-invitatory
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invitatory)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) pax)
+  [~ ~ %noun invitatory]
+::
+++  peek-x-invite
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invite)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) t.pax)
+  ?~  invitatory
+    ~
+  =/  uid=serial  (slav %uv i.pax)
```

Comment:
So a `/invite` scry path takes the form of `/invite/some-uid/invite-path`? Feels dirty to not have that go broad->specific. As in, following the data structures, `/invite/invite-path/some-uid`.

Reply:
A uid is fixed length and a path is variable length, so it makes more sense to prefix with uid than to suffix with it.

Reply:
I agree that makes sense from an _implementation_ standpoint, but would say that it doesn't from an _interface_ standpoint. It breaks convention of pretty much all path usage, both pre- and post-urbit.

Reply:
Most urbit paths follow the prefix convention or do weird things where they flop the paths.

Reply:
The path flopping is ancient and largely abandoned. And I only recall seeing it in comments, not in implementations. Reverse order is still better than arbitrary order though.

Reply:
It isn't arbitrary. It's specified.

Reply:
Scry paths are interfaces, and should be designed as such.

Reply:
Okay. Will change.

Reply:
Have changed this now.